Bottom: c01d8f1307fcab04129b21f5ab037212755a5469
Top:    f8a509b32e9de060516b4b095307e363c25abb51
Author: Ian Kent <ikent@redhat.com>
Date:   2013-04-04 17:18:24 +0800

Refresh of autofs-5.0.7-syncronize-handle_mounts-shutdown.patch

---

diff --git a/lib/master.c b/lib/master.c
index 6119d30..28c7717 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -1429,7 +1429,10 @@ again:
 	/* Try to reduce the number of times around the loop, waiting
 	 * for these guys to exit.
 	 */
-	sched_yield();
+	if (fc.busy) {
+		struct timespec tm = {0, 100000000};
+		nanosleep(&tm, NULL);
+	}
 
 	/*finish_mutex_lock();
 	error(LOGOPT_ANY, "after broadcast fc.busy %d", fc.busy);
